### ISSUE TITLE
[LLVM Analyzer] Invalid bitwise shift

### ISSUE THREAD

**Issue by iarspider** [cms-sw/cmssw#46037]
LLVM analyzer reports these instances of invalid bitwise shifts: 

* Shift by `2147483647` in [EgammaHLTPixelMatchVarProducer.cc](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-4221ac.html#EndPath)
* Negative shift in [`HOTriggerPrimitiveDigi::HOTriggerPrimitiveDigi`](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-801beb.html#EndPath), [`Stub::Stub`](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-4d0f82.html#EndPath), [`TCPReceiver::ReceiveLumiSection`](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-034bd4.html#EndPath)

(the last one could be a false-positive)

**Comment by iarspider** [comment]
assign DataFormats/HcalDigi, L1Trigger/TrackFindingTracklet, RecoEgamma/EgammaHLTProducers, RecoLuminosity/TCPReceiver

**Comment by iarspider** [comment]
assign L1Trigger/L1THGCal,EventFilter/HGCalRawToDigi

**Comment by mmusich** [comment]
> Shift by too-large value in [EgammaHLTPixelMatchVarProducer.cc](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-4221ac.html#EndPath),

tagging @Sam-Harper as original author of this code (via https://github.com/cms-sw/cmssw/pull/18541)

**Comment by kpedro88** [comment]
HOTriggerPrimitiveDigi: @jmmans 
TrackFindingTracklet/Stub: @skinnari @aryd

**Comment by mmusich** [comment]
>   * Shift by too-large value in [EgammaHLTPixelMatchVarProducer.cc](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-4221ac.html#EndPath)

I can't find this anymore in recent IBs (e.g. [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_15_0_X_2024-12-03-1100/el8_amd64_gcc12/llvm-analysis/)) I gess because of https://github.com/cms-sw/cmssw/pull/46488. @iarspider please confirm I am looking at the right place.

**Comment by iarspider** [comment]
> > * Shift by too-large value in [EgammaHLTPixelMatchVarProducer.cc](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_14_2_X_2024-09-16-1100/el8_amd64_gcc12/llvm-analysis/report-4221ac.html#EndPath)
> 
> I can't find this anymore in recent IBs (e.g. [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_15_0_X_2024-12-03-1100/el8_amd64_gcc12/llvm-analysis/)) I gess because of #46488. @iarspider please confirm I am looking at the right place.

Yes. The only remaining bit-shift warning is in [`hgcal::econd::produceERxData`](https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_15_0_X_2024-12-02-2300/el8_amd64_gcc12/llvm-analysis/report-f8f969.html#EndPath)

**Comment by mmusich** [comment]
+hlt
   * see https://github.com/cms-sw/cmssw/issues/46037#issuecomment-2514901945

**Comment by jfernan2** [comment]
+1

**Comment by kpedro88** [comment]
I no longer see these warnings in current IBs. should this issue be closed?

**Comment by smuzaffar** [comment]
yes , looks like the warning is no more there. Lets close this issue
